packages/gatsby-source-contentful/src/normalize.js

```diff
@@ -563,17 +519,39 @@ exports.createNodesForContentType = ({
           fieldType === `RichText` &&
           _.isPlainObject(entryItemFields[entryItemFieldKey])
         ) {
-          const richTextNode = prepareRichTextNode(
-            entryNode,
-            entryItemFieldKey,
-            entryItemFields[entryItemFieldKey],
-            createNodeId
-          )
+          const fieldValue = entryItemFields[entryItemFieldKey]
+
+          const rawReferences = []
+
+          // Locate all Contentful Links within the rich text data
+          const traverse = obj => {
+            for (let k in obj) {
+              const v = obj[k]
+              if (v && v.sys && v.sys.type === `Link`) {
+                rawReferences.push(v)
+              } else if (v && typeof v === `object`) {
+                traverse(v)
+              }
+            }
+          }
 
-          childrenNodes.push(richTextNode)
-          entryItemFields[`${entryItemFieldKey}___NODE`] = richTextNode.id
+          traverse(fieldValue)
 
-          delete entryItemFields[entryItemFieldKey]
+          // Build up resolvable reference list
+          const resolvableReferenceIds = rawReferences
+            .filter(function (v) {
+              return resolvable.has(
+                `${v.sys.id}___${v.sys.linkType || v.sys.type}`
+              )
+            })
+            .map(function (v) {
+              return mId(space.sys.id, v.sys.id, v.sys.linkType || v.sys.type)
+            })
+
+          entryItemFields[entryItemFieldKey] = {
+            raw: stringify(fieldValue),
+            [`references___NODE`]: resolvableReferenceIds,
```

Comment:
Is there a particular reason for using this syntax?
```suggestion
            references___NODE: resolvableReferenceIds,
```

Reply:
👍